src/main/java/com/volmit/iris/engine/object/IrisTreeSize.java

```diff
@@ -0,0 +1,188 @@
+package com.volmit.iris.engine.object;
+
+import com.volmit.iris.Iris;
+import com.volmit.iris.engine.object.annotations.Desc;
+import com.volmit.iris.util.collection.KList;
+import com.volmit.iris.util.collection.KMap;
+import org.bukkit.Location;
+import org.bukkit.Material;
+
+@Desc("Sapling override object picking options")
+public enum IrisTreeSize {
+
+    @Desc("Only one sapling")
+    ONE,
+
+    @Desc("Two by two area")
+    TWO,
+
+    @Desc("Three by three any location")
+    THREE_ANY,
+
+    @Desc("Three by three area with center")
+    THREE_CENTER,
+
+    @Desc("Four by four")
+    FOUR,
+
+    @Desc("Five by five")
+    FIVE_ANY,
+
+    @Desc("Five by five center")
+    FIVE_CENTER,
+
+    @Desc("Any size")
+    ANY;
+
+    /**
+     * All sizes in this enum
+     */
+    public static final KList<IrisTreeSize> sizes = new KList<>(ONE, TWO, THREE_ANY, THREE_CENTER, FOUR, FIVE_ANY, FIVE_CENTER);
+
+    /**
+     * Get the best size to match against from a list of sizes
+     * @param sizes The list of sizes
+     * @return The best size (highest & center > any)
+     */
+    public static IrisTreeSize bestSizeInSizes(KList<IrisTreeSize> sizes){
+        if (sizes.contains(ANY)){
+            return ANY;
+        }
+        else if (sizes.contains(FIVE_CENTER)){
+            return FIVE_CENTER;
+        }
+        else if (sizes.contains(FIVE_ANY)){
+            return FIVE_ANY;
+        }
+        else if (sizes.contains(FOUR)){
+            return FOUR;
+        }
+        else if (sizes.contains(THREE_CENTER)){
+            return THREE_CENTER;
+        }
+        else if (sizes.contains(THREE_ANY)){
+            return THREE_ANY;
+        }
+        else if (sizes.contains(TWO)){
+            return TWO;
+        }
+        else if (sizes.contains(ONE)){
+            return ONE;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Check if the size at a specific location is valid
+     * @param size the IrisTreeSize to check
+     * @param location at this location
+     * @return A list of locations if any match, or null if not.
+     */
+    public static KList<KList<Location>> getPlacesIfValid (IrisTreeSize size, Location location) {
+        if (size == ANY){
+            Iris.debug("ANY was passed to getPlacesIfValid while it should never!");
+        }
+        return switch (size){
+            case ONE            -> new KList<KList<Location>>(new KList<>(location));
+            case TWO            -> loopLocation(location, 2);
+            case THREE_ANY      -> loopLocation(location, 3);
+            case FOUR           -> loopLocation(location, 4);
+            case FIVE_ANY       -> loopLocation(location, 5);
+            case THREE_CENTER   -> isCenterMapValid(location, 3);
+            case FIVE_CENTER    -> isCenterMapValid(location, 5);
+            case ANY            -> null;
+        };
+    }
+
```

Comment:
Maybe use a more efficient way of finding a box of saplings? This uses a lot of memory (produces a lot of garbage) and could use a lot less cycles.

```java
public Cuboid getSaplings(Location at, Predicate<BlockData> valid)
{
    List<BlockPosition> v = new ArrayList<>();
    grow(at.getWorld(), new BlockPosition(at.getBlockX(), at.getBlockY(), at.getBlockZ()), valid, v);
    BlockPosition a = new BlockPosition(Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE);
    BlockPosition b = new BlockPosition(Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE);

    for(BlockPosition i : v)
    {
        a.max(i);
        b.min(i);
    }

    Cuboid c = new Cuboid(a.toBlock(at.getWorld()).getLocation(), b.toBlock(at.getWorld()).getLocation());
    boolean vx = true;

    while(Math.min(c.getSizeX(), c.getSizeZ()) > 2)
    {
        checking: for(int i = c.getLowerX(); i < c.getUpperX(); i++)
        {
            for(int j = c.getLowerY(); j < c.getUpperY(); j++)
            {
                for(int k = c.getLowerZ(); k < c.getUpperZ(); k++)
                {
                    BlockPosition p = new BlockPosition(i,j,k);

                    if(!v.contains(p))
                    {
                        vx = false;
                        break checking;
                    }
                }
            }
        }

        if(vx)
        {
            return c;
        }

        c.inset(Cuboid.CuboidDirection.Horizontal, 1);
        vx = true;
    }

    return null;
}

private boolean grow(World world, BlockPosition center, Predicate<BlockData> valid, List<BlockPosition> l)
{
    if(l.size() > 50)
    {
        return false;
    }

    if(!l.contains(center) && valid.test(center.toBlock(world).getBlockData()))
    {
        l.add(center);
        grow(world, center.add(1, 0, 0), valid, l);
        grow(world, center.add(-1, 0, 0), valid, l);
        grow(world, center.add(0, 0, 1), valid, l);
        grow(world, center.add(0, 0, -1), valid, l);
        return false;
    }

    return false;
}
```